### PR TITLE
K8s labels filters

### PR DESCRIPTION
**Pull request by arikalon1**

### CODE REVIEW

File: docs/catalog/triggers/kubernetes.rst [hunk 1]
```diff
@@ -46,3 +46,25 @@ The following wildcard triggers will fire for any supported Kubernetes resource:
...
 * on_kubernetes_any_resource_all_changes
+
+Supported filters
+--------------------
+
+All Kubernetes triggers support the following filters:
+
+* ``name_prefix`` - Name prefix to match resources.
+* ``namespace_prefix`` - Namespace prefix to match resources.
+* ``labels_selector`` - Match resources with these labels. The format is: ``label1=value1,label2=value2``. If more than one labels is provided, **all** needs to match.
```
Comment (aantn) [inline docs/catalog/triggers/kubernetes.rst:57]: Should be `need` not `needs`

File: docs/catalog/triggers/kubernetes.rst [hunk 2]
```diff
@@ -46,3 +46,25 @@ The following wildcard triggers will fire for any supported Kubernetes resource:
 * on_kubernetes_any_resource_update
 * on_kubernetes_any_resource_delete
 * on_kubernetes_any_resource_all_changes
+
```
Comment (aantn) [inline docs/catalog/triggers/kubernetes.rst:49]: Good job documenting it.

File: docs/catalog/triggers/kubernetes.rst [hunk 3]
```diff
@@ -46,3 +46,25 @@ The following wildcard triggers will fire for any supported Kubernetes resource:
...
+Supported filters
+--------------------
+
+All Kubernetes triggers support the following filters:
+
+* ``name_prefix`` - Name prefix to match resources.
+* ``namespace_prefix`` - Namespace prefix to match resources.
+* ``labels_selector`` - Match resources with these labels. The format is: ``label1=value1,label2=value2``. If more than one labels is provided, **all** needs to match.
+
+For example:
```
Comment (aantn) [inline docs/catalog/triggers/kubernetes.rst:59]: Can we put the example at the top of the page along with one sentence on the motivation?

I think we're jumping in too fast into the fine details without explaining the motivation. We can't assume that people read through the docs and reached this page in the order we expect.

See for example, what I did at https://docs.robusta.dev/master/catalog/triggers/prometheus.html

File: src/robusta/integrations/kubernetes/base_triggers.py [hunk 4]
```diff
@@ -40,6 +40,22 @@ class K8sBaseTrigger(BaseTrigger):
     operation: K8sOperationType = None
     name_prefix: str = None
     namespace_prefix: str = None
+    labels_selector: str = None
+    _labels_map: dict = PrivateAttr()
```
Comment (aantn) [inline src/robusta/integrations/kubernetes/base_triggers.py:44]: I think this has to be `Dict` (from `typing`) and not `dict` for compatibility with lower python versions

**Review by aantn — APPROVED** [review]
Looks good. Left some comments on the docs.

File: docs/catalog/triggers/kubernetes.rst (hunk 1 above)
Comment (arikalon1) [reply docs/catalog/triggers/kubernetes.rst:57]: fixing

File: docs/catalog/triggers/kubernetes.rst (hunk 3 above)
Comment (arikalon1) [reply docs/catalog/triggers/kubernetes.rst:59]: good idea. Fixing

File: src/robusta/integrations/kubernetes/base_triggers.py (hunk 4 above)
Comment (arikalon1) [reply src/robusta/integrations/kubernetes/base_triggers.py:44]: Fixing

**Pull request merged**